Use allocatable instead of capacity for node resources.

mcpappas opened the pull request:
Fixes #2097.

The Cluster and Node Memory metrics are not intuitive. The statistics shown are using kube_node_status_capacity instead of kube_node_status_allocatable which can be highly misleading for certain cloud providers/node VM sizes (due to provider specific resource reservations described here). In my circumstance using Azure Kubernetes Service with Standard_B2s results in a reported utilization of ~55% when it is actually ~98% according to kubectl describe node. This difference makes it very difficult to understand why a cluster would report insufficient memory when scheduling pods. Memory consumption relative to the cluster or node's total Allocatable Memory is a much more useful metric for most users who are generally only concerned with the resources available for consumption by pods/workloads. This is supported by kubectl's default reporting of node memory (i.e. relative to the allocatable memory).

nevalla reviewed (dismissed):
I think the changes are reasonable. LGTM.

BlackHole1 commented on src/renderer/components/+cluster/cluster-pie-charts.tsx:
```diff
@@ -45,20 +45,20 @@ export const ClusterPieCharts = observer(() => {
     const defaultColor = ThemeStore.getInstance().activeTheme.colors.pieChartDefaultColor;
-
-    if (!memoryCapacity || !cpuCapacity || !podCapacity) return null;
+    
```
Excess space

mcpappas committed:
Add allocatable metrics for node resources.
Display both Total Capacity and Allocatable Capacity on node and cluster graphs .

Signed-off-by: Michael Pappas <mhagen@mapcom.com>

mcpappas committed:
Make legends consistent, and correct workloadMemoryUsage query.

Signed-off-by: Michael Pappas <mhagen@mapcom.com>

mcpappas commented on src/renderer/components/+nodes/nodes.tsx:
```diff
@@ -97,7 +97,7 @@ export class Nodes extends React.Component<Props> {
 
   renderMemoryUsage(node: Node) {
-    const metrics = nodesStore.getLastMetricValues(node, ["memoryUsage", "memoryCapacity"]);
+    const metrics = nodesStore.getLastMetricValues(node, ["workloadMemoryUsage", "memoryAllocatableCapacity"]);
```
This is likely misleading when `(memoryAllocatableCapacity - memoryCapacity) < (memoryUsage - workloadMemoryUsage)`. I considered revising it, but I am pressed for time.

Nokel81 replied:
What do you think it should be instead during when that is true?

mcpappas replied:
I think my first comment had a mistake (was rushing to catch a train). When `memoryCapacity - memoryAllocatableCapacity < memoryUsage - workloadMemoryUsage` Then it probably makes more sense to compare `memoryUsage - (memoryCapacity - memoryAllocatableCapacity)` to `memoryAllocatableCapacity`.

That probably seems a bit like magic math, so to be clear: 
`(memoryCapacity - memoryAllocatableCapacity)` represents the portion of memory that is reserved for non-workloads (kubelet, etc.). If those “non-workloads” exceed the reserved amount (or there was no reservation at all) then they begin to consume `memoryAllocatableCapacity`.

Nokel81 replied:
Ah that makes sense. But can be done in the future.

mcpappas replied:
Agreed, the other charts should provide sufficient clarity for now, and would be a small change.

Nokel81 reviewed (approved):
Thank you very much for this improvement. If you want to improve it further down the line. More PRs would be very much appreciated.
